.github/workflows/tests_emulator.yml

```
@@ -158,6 +158,28 @@ jobs:
             sleep 5
             ./gradlew uninstallAll jacocoAndroidTestReport --daemon
 
+      - name: Install NDK tools
+        run: |
+          echo "Installing NDK tools for alignment checking"
+          yes | sdkmanager --install "ndk;27.2.12479018"
+          echo "export PATH=$ANDROID_HOME/ndk/27.2.12479018/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH" >> $GITHUB_ENV
+
+      - name: Check APK ELF Alignment
+        run: |
+          chmod +x .github/check_elf_alignment.sh
+          if [ ! -d "AnkiDroid/build/outputs/apk/" ]; then
+            ./gradlew :AnkiDroid:assemblePlayRelease --daemon
+          fi
+          APK_PATH="AnkiDroid/build/outputs/apk/play/release/AnkiDroid-play-arm64-v8a-release.apk"
```

Is this specific APK always expected to be generated from jacocoAndroidTestReport in CI (which runs in release mode, I think Play flavor) ?

If so, I think it's better to just fail here with a message saying exactly what filename is missing rather than try to rebuild it out of the regular build/test context above. Mostly because it takes some special environment variable setup and keystore / key setup for it to work and I'd like to only think about that in one build spot in the workflow

---

Yep it is guaranteed to be generated since it is the same one generated for compare_apk_size.yml
https://github.com/ankidroid/Anki-Android/blob/c719bd4f525cba28297aa61b88d48c69b3603cbe/.github/workflows/compare_apk_size.yml#L67

---

> So remove ELF Alignment check? Google does recommend checking for it as part of the 16kb page alignment process.

That's not what I said at all. I said

> If [this specific APK always expected to be generated from jacocoAndroidTestReport], I think it's better to just fail here with a message saying exactly what filename is missing rather than try to rebuild it

I don't want to attempt to rebuild-if-missing here. I want to assume it's already built (it should already be built?) and fail fast if not.

---

Ohh, sorry about that, I assumed it wasn't already being built. Yeah that sounds good!